agent/replacecall/src/main/java/randoop/instrument/ReplaceCallAgent.java

```diff
@@ -0,0 +1,201 @@
+      } catch (IOException e) {
+        System.err.format(
+            "Error reading package exclusion file %s:%n %s%n", dont_transform, e.getMessage());
+        System.exit(1); // Exit on user input error. (Throwing exception would halt JVM.)
+      }
+    }
+
+    /*
+     * The agent is called when classes are loaded. If Randoop is using threads, this can result in
+     * multiple threads accessing the map to apply replacements. Since
+     */
+    ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap = new ConcurrentHashMap<>();
+
+    // Read the default replacement file
+    String replacementPath = "/default-replacements.txt";
+    inputStream = ReplaceCallAgent.class.getResourceAsStream(replacementPath);
+    if (inputStream == null) {
+      System.err.println("Unable to open default replacements file. Please report.");
+      System.exit(1);
+    }
+    try {
+      replacementMap =
+          ReplacementFileReader.readReplacements(
+              new InputStreamReader(inputStream), replacementPath);
+    } catch (ReplacementFileException e) {
+      System.err.printf("Error reading default replacement file:%n  %s%n", e);
+      System.err.println("Check that replacecall.jar is on the classpath or bootclasspath.");
+      System.exit(1);
+    }
+
+    // If the user has provided a replacement file, load user replacements and put them into the
+    // map, possibly overriding default replacements that already appear in the map.
+    if (map_calls != null) {
+      try {
+        replacementMap.putAll(ReplacementFileReader.readReplacements(map_calls));
+      } catch (Throwable e) {
+        System.err.printf("Error reading replacement file %s:%n  %s%n", map_calls, e.getMessage());
+        System.exit(1);
+      }
+    }
+
+    // Communicate the list of replaced methods to Randoop to omit direct calls
+    List<String> signatureList = new ArrayList<>();
+    for (MethodSignature def : replacementMap.keySet()) {
+      signatureList.add(def.toString());
+    }
+    MethodReplacements.addReplacedMethods(signatureList);
+
+    CallReplacementTransformer transformer =
+        new CallReplacementTransformer(replacementMap, excludedPackagePrefixes);
+    transformer.addMapFileShutdownHook();
+
+    inst.addTransformer(transformer);
+  }
+
+  /**
+   * Load package names from the given file and add them to the set of excluded package names. Adds
+   * a period to the end of any name that does not have one.
+   *
+   * @param exclusionReader the reader for the text file containing the list of excluded packages,
+   *     must not be null
+   * @param filename the name of the file read by the reader
+   * @param excludedPackagePrefixes the set of excluded package prefixes, modified by adding new
+   *     prefixes. Must not be null.
+   * @throws IOException if there is an error reading the file
+   */
+  private static void loadExclusions(
```

Comment:
Any method contract that side-effects parameters is confusing to understand and requires a reader to scan large amounts of the code to understand all the places that side effects happen.  Please avoid this coding style.
In this case, it is unnecessary.  Just have loadExclusions return a set.  Then, `excludedPackagePrefixes` can be set from the output of loadExclusions rather than passed in for side effect.  Both the implementation of this method, and the client of this method, will become simpler and clearer.